docs/rules-engine/README.md

```diff
@@ -0,0 +1,149 @@
+>
+> The special characters used in the pattern should contain a double backslash (`\\`).
+> For example, to check if a string contains a `\t`, the pattern would need to include `\\t`.
+>
+> Also, to avoid the wrong detection of an ES RegExp instead of RegExp content, a content beginning with a slash `/` character
+> (such as a path `/path/to/file`) should be preceded by a double backslash `\\` (for example `\\/path/to/file`)
+
+### Action
+
+The ActionBlock defines the action to be executed. 
+Each Action is identified by a unique ID, specifying the process to trigger and any associated payload.
+
+The Otter framework provides a few built-in actions linked to Otter-specific features:
+* `UPDATE_CONFIG` modifies the configuration of a configurable component
+(provided by [@o3r/configuration](https://www.npmjs.com/package/@o3r/configuration))
+* `UPDATE_ASSET` overrides the path of an asset which uses the `o3rDynamicContent` pipe 
+(provided by [@o3r/dynamic-content](https://www.npmjs.com/package/@o3r/dynamic-content))
+* `UPDATE_LOCALISATION` substitutes the translation of a key with the one from another key.
+([@o3r/localization](https://www.npmjs.com/package/@o3r/localization))
+* `UPDATE_PLACEHOLDER` injects HTMLcode in a specific placeholder
+([@o3r/components](https://www.npmjs.com/package/@o3r/components))
+
+You can provide your own custom action following [this guide](./how-to-use/custom-action.md).
+
+> [!NOTE]
+> The actions handled by the rules engine need to be [registered](./how-to-use/README.md#step-2-register-the-action-handlers-in-the-runner)
+> in your application to be recognized by the engine.
+> This ensures you only import packages you actually need.
+
+### Fact
+
+A fact is a stream of values.
+
+During a rule execution, its conditions will be evaluated using the latest fact value(s).
+Every fact can be identified by its `id` property and exposes a `value$` stream that will trigger a new evaluation
+of the Ruleset it has been linked for every value it emits.
+This allows a big optimization since the rules engine will react to a fact change in order to execute only the subset
+of rules that relies on this fact.
+
+Facts can be defined inside the code of the application, as described in the [custom fact documentation](./how-to-use/custom-fact.md),
+or can be defined by the ruleset itself. 
+
+In that case, we call it a [runtime fact](#runtime-fact).
+
+> [!WARNING]
+> A rule depending on a fact will be re-evaluated whenever this fact is updated. 
+> If you use a fact that will be updated a lot in a short time frame, you may encounter performance issues.
+> 
+> You can avoid this by:
+> * disabling the ruleset if the component impacted by the rule is missing from the page (see [linkedComponents](./how-to-use/dedicated-component-ruleset.md))
+> * avoiding using [JSONPath](./examples/complex-fact.md) on facts that are updated a lot in a short time frame and
```

Comment:
should now be less a problem with the distinctUntilChanged : 
![image](https://github.com/AmadeusITGroup/otter/assets/122815580/31c19b3e-befa-4643-b0a1-61be091fd9c8)

Reply:
But still a good practice, and if a subset is used several times it's better to create a dedicated fact instead of applying the same jsonPath several times on the same fact.

Reply:
As discussed, I will remove it as it is sensibly the same (JSONPath versus map / pick operator that will be evaluated as many times)